### PR TITLE
feat(models): Added ExternalProjectIntegration Model

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

**Comment by lauryndbrown** [comment]
Note since this will not connect with the `ExternalIssue` table, when we are syncing statuses for Sentry -> Jira/VSTS we will need to make another api call in order to see which project an issue is from.

File: src/sentry/models/externalintegrationproject.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class ExternalProjectIntegration(Model):
+    __core__ = False
+
+    organization_integration_id = BoundedPositiveIntegerField(db_index=True)
+    date_added = models.DateTimeField(default=timezone.now)
+    name = models.CharField(max_length=128)
+    external_id = models.CharField(max_length=64)
+    resolved_status = models.CharField(max_length=64)
+    unresolved_status = models.CharField(max_length=64)
```
Comment (lauryndbrown) [inline src/sentry/models/externalintegrationproject.py]: Wasn't sure about the lengths on these. Any suggestions?

Comment (mattrobenolt) [reply src/sentry/models/externalintegrationproject.py]: In Postgres, a text column, and a varchar are technically the same thing. You can think of a varchar simply as a text + a constraint. There's no efficiency gain by using a varchar aside from enforcing a length. So if you don't particularly care, you can just feel free to make them `TextField()` and not worry about the length.

If it's user import, can alwyas just restrict the length in the form validation or input data validation at that point, and keep it looser in the database. It'll avoid needing to change it later if say, 64 characters is not enough, etc.

Just generic knowledge sharing, I don't have a lot of context on what these are. :)

File: src/sentry/models/externalintegrationproject.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from __future__ import absolute_import
```
[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/models/externalintegrationproject.py]: yeah agree that's a better naming/organization strategy

File: src/sentry/models/externalintegrationproject.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    organization_integration_id = BoundedPositiveIntegerField(db_index=True)
+    date_added = models.DateTimeField(default=timezone.now)
+    name = models.CharField(max_length=128)
+    external_id = models.CharField(max_length=64)
+    resolved_status = models.CharField(max_length=64)
+    unresolved_status = models.CharField(max_length=64)
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_externalprojectintegration'
```
Comment (macqueen) [inline src/sentry/models/externalintegrationproject.py]: make sure you update this too if you're changing the model name

**Review by macqueen — COMMENTED** [review]
don't forget to generate the migration

**Comment by macqueen** [comment]
oops didn't see this comment 😬 

> Needs a generated migration as well :)

**Review by macqueen — COMMENTED** [review]
it looks like you're creating both the IntegrationExternalProject and ExternalProjectIntegration models in your migration file. you can probably avoid that happening by making sure you delete pyc files before generating these

**Review by macqueen — APPROVED** [review]

**Pull request merged**